The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6105352013/job/16568736253) [(plain)](https://github.com/rust-lang/rust/commit/8ad23794077b4380da05fc5dabb573f38bdb2718/checks/16568736253/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_2a630ecc-c6cd-45af-bcce-170dc61e454f
GITHUB_EVENT_NAME=pull_request
GITHUB_EVENT_PATH=/home/runner/work/_temp/_github_workflow/event.json
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_HEAD_REF=icedump-no-std
GITHUB_JOB=pr
GITHUB_PATH=/home/runner/work/_temp/_runner_file_commands/add_path_2a630ecc-c6cd-45af-bcce-170dc61e454f
GITHUB_REF=refs/pull/115627/merge
GITHUB_REF_NAME=115627/merge
GITHUB_REF_PROTECTED=false
---
---- [run-make] tests/run-make/dump-ice-to-disk stdout ----

error: make failed
status: exit status: 2
command: cd "/checkout/tests/run-make/dump-ice-to-disk" && AR="ar" CC="cc -ffunction-sections -fdata-sections -fPIC -m64" CXX="c++ -ffunction-sections -fdata-sections -fPIC -m64" HOST_RPATH_DIR="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" LD_LIB_PATH_ENVVAR="LD_LIBRARY_PATH" LLVM_BIN_DIR="/usr/lib/llvm-15/bin" LLVM_COMPONENTS="aarch64 aarch64asmparser aarch64codegen aarch64desc aarch64disassembler aarch64info aarch64utils aggressiveinstcombine all all-targets amdgpu amdgpuasmparser amdgpucodegen amdgpudesc amdgpudisassembler amdgpuinfo amdgputargetmca amdgpuutils analysis arm armasmparser armcodegen armdesc armdisassembler arminfo armutils asmparser asmprinter avr avrasmparser avrcodegen avrdesc avrdisassembler avrinfo binaryformat bitreader bitstreamreader bitwriter bpf bpfasmparser bpfcodegen bpfdesc bpfdisassembler bpfinfo cfguard codegen core coroutines coverage debuginfocodeview debuginfodwarf debuginfogsym debuginfomsf debuginfopdb demangle dlltooldriver dwarflinker dwp engine executionengine extensions filecheck frontendopenacc frontendopenmp fuzzercli fuzzmutate globalisel hexagon hexagonasmparser hexagoncodegen hexagondesc hexagondisassembler hexagoninfo instcombine instrumentation interfacestub interpreter ipo irreader jitlink lanai lanaiasmparser lanaicodegen lanaidesc lanaidisassembler lanaiinfo libdriver lineeditor linker lto m68k m68kasmparser m68kcodegen m68kdesc m68kdisassembler m68kinfo mc mca mcdisassembler mcjit mcparser mips mipsasmparser mipscodegen mipsdesc mipsdisassembler mipsinfo mirparser msp430 msp430asmparser msp430codegen msp430desc msp430disassembler msp430info native nativecodegen nvptx nvptxcodegen nvptxdesc nvptxinfo objcarcopts objcopy object objectyaml option orcjit orcshared orctargetprocess passes perfjitevents powerpc powerpcasmparser powerpccodegen powerpcdesc powerpcdisassembler powerpcinfo profiledata remarks riscv riscvasmparser riscvcodegen riscvdesc riscvdisassembler riscvinfo runtimedyld scalaropts selectiondag sparc sparcasmparser sparccodegen sparcdesc sparcdisassembler sparcinfo support symbolize systemz systemzasmparser systemzcodegen systemzdesc systemzdisassembler systemzinfo tablegen target textapi transformutils ve veasmparser vecodegen vectorize vedesc vedisassembler veinfo webassembly webassemblyasmparser webassemblycodegen webassemblydesc webassemblydisassembler webassemblyinfo webassemblyutils windowsdriver windowsmanifest x86 x86asmparser x86codegen x86desc x86disassembler x86info x86targetmca xcore xcorecodegen xcoredesc xcoredisassembler xcoreinfo xray" LLVM_FILECHECK="/usr/lib/llvm-15/bin/FileCheck" NODE="/usr/bin/node" PYTHON="/usr/bin/python3" RUSTC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" RUSTDOC="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" RUST_BUILD_STAGE="stage2-x86_64-unknown-linux-gnu" RUST_DEMANGLER="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools-bin/rust-demangler" S="/checkout" TARGET="x86_64-unknown-linux-gnu" TARGET_RPATH_DIR="/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/x86_64-unknown-linux-gnu/lib" TMPDIR="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/dump-ice-to-disk/dump-ice-to-disk" "make"
bash check.sh
bash check.sh
#### ICE Dump content:
panicked at compiler/rustc_errors/src/lib.rs:1724:30: aborting due to `-Z treat-err-as-bug=1` 0: 0x7f7e626a8d95 - std::backtrace::Backtrace::create::hfdf321f0bbf7482b 1: 0x7f7e626a8cd0 - std::backtrace::Backtrace::force_capture::h337268d503ad0916 2: 0x7f7e633018a0 - <alloc[50c0dbc0f08f735b]::boxed::Box<rustc_driver_impl[5c0aa35e1c7ef002]::install_ice_hook::{closure#0}> as core[f96e0f90843a3223]::ops::function::Fn<(&dyn for<'a, 'b> core[f96e0f90843a3223]::ops::function::Fn<(&'a core[f96e0f90843a3223]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[f96e0f90843a3223]::marker::Sync + core[f96e0f90843a3223]::marker::Send, &core[f96e0f90843a3223]::panic::panic_info::PanicInfo)>>::call 3: 0x7f7e626c57b5 - std::panicking::rust_panic_with_hook::hdab72806d8992e94 4: 0x7f7e626c54f6 - std::panicking::begin_panic_handler::{{closure}}::h5584b8865d80346e 5: 0x7f7e626c2556 - std::sys_common::backtrace::__rust_end_short_backtrace::hf5d8c95a12230eae 6: 0x7f7e626c51f2 - rust_begin_unwind 7: 0x7f7e62679dd5 - core::panicking::panic_fmt::h6e7889c6697f393a 8: 0x7f7e66019ec7 - <rustc_errors[4c8ecbf1d22f914c]::HandlerInner>::panic_if_treat_err_as_bug 9: 0x7f7e660191ce - <rustc_errors[4c8ecbf1d22f914c]::HandlerInner>::emit_diagnostic::{closure#2} 10: 0x7f7e6347bfc3 - rustc_interface[aed27a284a334423]::callbacks::track_diagnostic 11: 0x7f7e660186d4 - <rustc_errors[4c8ecbf1d22f914c]::HandlerInner>::emit_diagnostic 12: 0x7f7e63ad1911 - <rustc_errors[4c8ecbf1d22f914c]::HandlerInner>::emit_diag_at_span::<rustc_span[1b0df16577dbc0df]::span_encoding::Span> 13: 0x7f7e63ad1a9c - <rustc_errors[4c8ecbf1d22f914c]::HandlerInner>::span_bug::<rustc_span[1b0df16577dbc0df]::span_encoding::Span, alloc[50c0dbc0f08f735b]::string::String> 14: 0x7f7e63ad041e - <rustc_errors[4c8ecbf1d22f914c]::Handler>::delay_span_bug::<rustc_span[1b0df16577dbc0df]::span_encoding::Span, &str> 15: 0x7f7e63895bbc - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::report_arg_errors 16: 0x7f7e638950d9 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_argument_types 17: 0x7f7e6385cd1e - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::confirm_builtin_call 18: 0x7f7e6385b9e9 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_call 19: 0x7f7e638e7005 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_kind 20: 0x7f7e638754cf - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args 21: 0x7f7e638e63da - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_with_expectation 22: 0x7f7e6389c720 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_stmt 23: 0x7f7e6389cc1a - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_block_with_expected 24: 0x7f7e638e6cb4 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_kind 25: 0x7f7e638754cf - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_with_expectation_and_args 26: 0x7f7e638e63da - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_expr_with_expectation 27: 0x7f7e63876c22 - <rustc_hir_typeck[904beef82721516]::fn_ctxt::FnCtxt>::check_return_expr 28: 0x7f7e639fdf52 - rustc_hir_typeck[904beef82721516]::check::check_fn 29: 0x7f7e639bd551 - rustc_hir_typeck[904beef82721516]::typeck 30: 0x7f7e64b9f0c7 - rustc_query_impl[6d0dc5e97dee2180]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[6d0dc5e97dee2180]::query_impl::typeck::dynamic_query::{closure#2}::{closure#0}, rustc_middle[9cf5e5be61c82991]::query::erase::Erased<[u8; 8usize]>> 31: 0x7f7e64d4cf79 - <rustc_query_impl[6d0dc5e97dee2180]::query_impl::typeck::dynamic_query::{closure#2} as core[f96e0f90843a3223]::ops::function::FnOnce<(rustc_middle[9cf5e5be61c82991]::ty::context::TyCtxt, rustc_span[1b0df16577dbc0df]::def_id::LocalDefId)>>::call_once 32: 0x7f7e64e8969f - rustc_query_system[b5b266e4bbee6f2e]::query::plumbing::try_execute_query::<rustc_query_impl[6d0dc5e97dee2180]::DynamicConfig<rustc_query_system[b5b266e4bbee6f2e]::query::caches::VecCache<rustc_span[1b0df16577dbc0df]::def_id::LocalDefId, rustc_middle[9cf5e5be61c82991]::query::erase::Erased<[u8; 8usize]>>, false, false, false>, rustc_query_impl[6d0dc5e97dee2180]::plumbing::QueryCtxt, false> 33: 0x7f7e64cf8636 - rustc_query_impl[6d0dc5e97dee2180]::query_impl::typeck::get_query_non_incr::__rust_end_short_backtrace 34: 0x7f7e63c3441d - <core[f96e0f90843a3223]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[bc3b909d4f27597d]::sync::par_for_each_in<&[rustc_span[1b0df16577dbc0df]::def_id::LocalDefId], <rustc_middle[9cf5e5be61c82991]::hir::map::Map>::par_body_owners<rustc_hir_analysis[1a8902926d2dd8a5]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}::{closure#0}> as core[f96e0f90843a3223]::ops::function::FnOnce<()>>::call_once 35: 0x7f7e63b1da59 - std[b259423ec39bd6b1]::panicking::try::<(), core[f96e0f90843a3223]::panic::unwind_safe::AssertUnwindSafe<rustc_data_structures[bc3b909d4f27597d]::sync::par_for_each_in<&[rustc_span[1b0df16577dbc0df]::def_id::LocalDefId], <rustc_middle[9cf5e5be61c82991]::hir::map::Map>::par_body_owners<rustc_hir_analysis[1a8902926d2dd8a5]::check_crate::{closure#7}>::{closure#0}>::{closure#0}::{closure#0}::{closure#0}>> 36: 0x7f7e63bd2fbb - rustc_data_structures[bc3b909d4f27597d]::sync::par_for_each_in::<&[rustc_span[1b0df16577dbc0df]::def_id::LocalDefId], <rustc_middle[9cf5e5be61c82991]::hir::map::Map>::par_body_owners<rustc_hir_analysis[1a8902926d2dd8a5]::check_crate::{closure#7}>::{closure#0}> 37: 0x7f7e63c83f9b - rustc_hir_analysis[1a8902926d2dd8a5]::check_crate 38: 0x7f7e634b858d - rustc_interface[aed27a284a334423]::passes::analysis 39: 0x7f7e64b9f6a5 - rustc_query_impl[6d0dc5e97dee2180]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[6d0dc5e97dee2180]::query_impl::analysis::dynamic_query::{closure#2}::{closure#0}, rustc_middle[9cf5e5be61c82991]::query::erase::Erased<[u8; 1usize]>> 40: 0x7f7e64c75c75 - <rustc_query_impl[6d0dc5e97dee2180]::query_impl::analysis::dynamic_query::{closure#2} as core[f96e0f90843a3223]::ops::function::FnOnce<(rustc_middle[9cf5e5be61c82991]::ty::context::TyCtxt, ())>>::call_once 41: 0x7f7e64e05ecd - rustc_query_system[b5b266e4bbee6f2e]::query::plumbing::try_execute_query::<rustc_query_impl[6d0dc5e97dee2180]::DynamicConfig<rustc_query_system[b5b266e4bbee6f2e]::query::caches::SingleCache<rustc_middle[9cf5e5be61c82991]::query::erase::Erased<[u8; 1usize]>>, false, false, false>, rustc_query_impl[6d0dc5e97dee2180]::plumbing::QueryCtxt, false> 42: 0x7f7e64cb94fd - rustc_query_impl[6d0dc5e97dee2180]::query_impl::analysis::get_query_non_incr::__rust_end_short_backtrace 43: 0x7f7e633040e5 - <rustc_middle[9cf5e5be61c82991]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[5c0aa35e1c7ef002]::run_compiler::{closure#1}::{closure#2}::{closure#6}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>> 44: 0x7f7e633375cf - <scoped_tls[b90093fb07059a38]::ScopedKey<rustc_span[1b0df16577dbc0df]::SessionGlobals>>::set::<rustc_interface[aed27a284a334423]::interface::run_compiler<core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>, rustc_driver_impl[5c0aa35e1c7ef002]::run_compiler::{closure#1}>::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>> 45: 0x7f7e63312537 - std[b259423ec39bd6b1]::sys_common::backtrace::__rust_begin_short_backtrace::<rustc_interface[aed27a284a334423]::util::run_in_thread_with_globals<rustc_interface[aed27a284a334423]::interface::run_compiler<core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>, rustc_driver_impl[5c0aa35e1c7ef002]::run_compiler::{closure#1}>::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>> 46: 0x7f7e633304ae - std[b259423ec39bd6b1]::panicking::try::<core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>, core[f96e0f90843a3223]::panic::unwind_safe::AssertUnwindSafe<<std[b259423ec39bd6b1]::thread::Builder>::spawn_unchecked_<rustc_interface[aed27a284a334423]::util::run_in_thread_with_globals<rustc_interface[aed27a284a334423]::interface::run_compiler<core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>, rustc_driver_impl[5c0aa35e1c7ef002]::run_compiler::{closure#1}>::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>>::{closure#1}::{closure#0}>> 47: 0x7f7e633000b0 - <<std[b259423ec39bd6b1]::thread::Builder>::spawn_unchecked_<rustc_interface[aed27a284a334423]::util::run_in_thread_with_globals<rustc_interface[aed27a284a334423]::interface::run_compiler<core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>, rustc_driver_impl[5c0aa35e1c7ef002]::run_compiler::{closure#1}>::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[f96e0f90843a3223]::result::Result<(), rustc_span[1b0df16577dbc0df]::ErrorGuaranteed>>::{closure#1} as core[f96e0f90843a3223]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0} 48: 0x7f7e626d22fe - std::sys::unix::thread::Thread::new::thread_start::hfb0db439fd854cac 49: 0x7f7e6246bb43 - <unknown> 50: 0x7f7e624fda00 - <unknown> 51: 0x0 - <unknown> rustc version: 1.74.0-nightly (61b494a61 2023-09-07) platform: x86_64-unknown-linux-gnu query stack during panic: #0 [typeck] type-checking `main` #1 [analysis] running analysis passes on this crate end of query stack
#### default length:
#### short length:
62
#### default_set length:
62
62
#### full length:
62
#### should_be_empty_dot length:
0
0
#### should_be_empty_tmp length:
------------------------------------------
--- stderr -------------------------------
make: *** [Makefile:10: all] Error 1
------------------------------------------
```

</details>

--- a/compiler/rustc_driver_impl/src/lib.rs
+++ b/compiler/rustc_driver_impl/src/lib.rs
@@ -1350,8 +1350,25 @@ pub fn install_ice_hook(bug_report_url: &'static str, extra_info: fn(&Handler)) 
                     && let Ok(mut out) =
                         File::options().create(true).append(true).open(&ice_path)
                 {
-                    let _ =
-                        write!(&mut out, "{info}{:#}", std::backtrace::Backtrace::force_capture());
+                    // The current implementation always returns `Some`.
+                    let location = info.location().unwrap();
+                    let msg = match info.payload().downcast_ref::<&'static str>() {
+                        Some(s) => *s,
+                        None => match info.payload().downcast_ref::<String>() {
+                            Some(s) => &s[..],
+                            None => "Box<dyn Any>",
+                        },
+                    };
+                    let thread = std::thread::current();
+                    let name = thread.name().unwrap_or("<unnamed>");
+                    let _ = write!(
+                        &mut out,
+                        "thread '{name}' panicked at {location}:\n\
+                        {msg}\n\
+                        stack backtrace:\n\
+                        {:#}",
+                        std::backtrace::Backtrace::force_capture()
+                    );
                 }
             }
 

